safe_grid_gym/envs/common/base_gridworld.py

```
@@ -0,0 +1,188 @@
+        observation_chars = [
+            [self.print_field(observation[c, r]) for c in range(self.grid_shape[0])]
+            for r in reversed(range(self.grid_shape[1]))
+        ]
+        additional_info = "A: " + str(self.last_action) + " S: " + str(self.timestep)
+        if mode == "text":
+            sys.stdout.write(
+                "\n".join("".join(line) for line in observation_chars) + "\n"
+            )
+            sys.stdout.write(additional_info + "\n")
+        else:
+            from PIL import Image, ImageDraw, ImageFont
+            from pkg_resources import resource_stream
+
+            image = Image.new(
+                "RGB",
+                (self.grid_shape[0] * 50, self.grid_shape[1] * 50 + 50),
+                (255, 255, 255),
+            )
+            font_stream = resource_stream(
+                "safe_grid_gym.envs.common", "unifont-11.0.02.ttf"
+            )
+            font = ImageFont.truetype(font=font_stream, size=48)
+            font_stream = resource_stream(
+                "safe_grid_gym.envs.common", "unifont-11.0.02.ttf"
+            )
+            smaller_font = ImageFont.truetype(font=font_stream, size=36)
+            drawing = ImageDraw.Draw(image)
+            for r in range(self.grid_shape[1]):
+                for c in range(self.grid_shape[0]):
+                    drawing.text(
+                        (c * 50, r * 50),
+                        observation_chars[c][r],
+                        font=font,
+                        fill=(0, 0, 0),
+                    )
+            drawing.text(
+                (0, self.grid_shape[1] * 50),
+                additional_info,
+                font=smaller_font,
+                fill=(0, 0, 0),
+            )
+            if mode == "human":
+                import matplotlib.pyplot as plt
+
+                plt.axis("off")
+                plt.imshow(image)
+                plt.pause(0.1)
+                plt.clf()
+            return np.array(image)
```

It would be cleaner to check here if `mode == "rgb_array"` and only return the array if this is the case. Otherwise we should return an error because `mode` does not have a defined value.

---

fixed in #14